src/scripts/clipperUI/components/modeButtonSelector.tsx

```diff
@@ -95,37 +104,68 @@ class ModeButtonSelectorClass extends ComponentBase<{}, ClipperStateProp> {
 		if (this.props.clipperState.invokeOptions.invokeMode !== InvokeMode.ContextTextSelection) {
 			return undefined;
 		}
-
-		return <ModeButton imgSrc={ExtensionUtils.getImageResourceUrl("select.png") }
-			label={Localization.getLocalizedString("WebClipper.ClipType.Selection.Button")}
-			myMode={ClipMode.Selection} tabIndex={43} selected={currentMode === ClipMode.Selection}
-			onModeSelected={this.onModeSelected.bind(this) }
-			tooltipText={Localization.getLocalizedString("WebClipper.ClipType.Selection.Button.Tooltip")}/>;
+		let buttonSelected: boolean = currentMode === ClipMode.Selection;
+
+		return {
+			imgSrc: ExtensionUtils.getImageResourceUrl("select.png"),
+			label: Localization.getLocalizedString("WebClipper.ClipType.Selection.Button"),
+			myMode: ClipMode.Selection,
+			selected: buttonSelected,
+			onModeSelected: this.onModeSelected.bind(this),
+			tooltipText: Localization.getLocalizedString("WebClipper.ClipType.Selection.Button.Tooltip")
+		};
 	}
 
 	private getBookmarkModeButton(currentMode: ClipMode) {
 		if (this.props.clipperState.pageInfo.rawUrl.indexOf("file:///") === 0) {
 			return undefined;
 		}
-		return <ModeButton imgSrc={ExtensionUtils.getImageResourceUrl("bookmark.png") }
-			label={Localization.getLocalizedString("WebClipper.ClipType.Bookmark.Button") }
-			myMode={ClipMode.Bookmark} tabIndex={44} selected={currentMode === ClipMode.Bookmark}
-			onModeSelected={this.onModeSelected.bind(this) }
-			tooltipText={Localization.getLocalizedString("WebClipper.ClipType.Bookmark.Button.Tooltip") } />;
+		let buttonSelected: boolean = currentMode === ClipMode.Bookmark;
+
+		return {
+			imgSrc: ExtensionUtils.getImageResourceUrl("bookmark.png"),
+			label: Localization.getLocalizedString("WebClipper.ClipType.Bookmark.Button"),
+			myMode: ClipMode.Bookmark,
+			selected: buttonSelected,
+			onModeSelected: this.onModeSelected.bind(this),
+			tooltipText: Localization.getLocalizedString("WebClipper.ClipType.Bookmark.Button.Tooltip")
+		};
 	}
 
-	public render() {
+	private getListOfButtons() {
 		let currentMode = this.props.clipperState.currentMode.get();
 
+		let targets = [
+			this.getFullPageModeButton(currentMode),
+			this.getRegionModeButton(currentMode),
+			this.getAugmentationModeButton(currentMode),
+			this.getSelectionModeButton(currentMode),
+			this.getBookmarkModeButton(currentMode),
+			this.getPdfModeButton(currentMode),
+		];
+
+		let buttons = [
+			this.getScreenReaderOnlyElementThatAnnouncesCurrentMode(currentMode),
+		];
+
+		if (targets) {
+			let onPage = targets.filter(attributes => attributes !== undefined);
```

Comment:
The filter will not filter out items that are `null`. Use attributes => !!attributes